automation.md

```diff
@@ -0,0 +1,75 @@
+# Automation
+
+## Overview
+
+The `IotTelemtrySimulator.Automation` .NET Standards 2.1 library allows you to run the IoT Telemetry Simulator as part of a pipeline or any other automation framework.
+
+It authenticates with Azure either using managed identities or a service principal, sets up the Azure resource group, container group(s) and container instance(s) needed, pulls the IotTelemetrySimulator Docker container from an Azure container registry, configures and runs the simulator, waits for it to finish, pulls and outputs it's log files and then tears down the Container instances and groups and the resource group.
+
+![IotTelemetrySimulator Automation Architecture](/docs/images/iottelemetrysimulator_automation_architecture.png)
+
+This way, automated load tests can be run as part of a CD/CI pipeline or any automation framework that allows executing arbitrary code.
+
+## Project Structure
+
+[IotTelemetrySimulator.Atuomation](/src/IotTelemetrySimulator.Automation) : The .NET Standards 2.1 automation library.
+
+* [appsettings.json](/src/IotTelemetrySimulator.Automation/appsettings.json) : Configuration file.
+
+[IotTelemetrySimulator.Atuomation.Console](/src/IotTelemetrySimulator.Automation.Console) : A .NET Core 3.1 console app that illustrates how to use the automation component.
+
+## Setup
+
+If you chose *Service Principal* authentication, you need to create an Active Directory application with the rights to create and remove resources in the desired Azure subscription. To achieve this, create a new application in Azure Active Directory and then navigate to "Subscriptions &rarr; [Subscription Id] &rarr; Access Control (IAM) &rarr; Role Assignments" and then add the newly created AAD Application with "Contributor" role.
+
+If you chose *Managed Identity* authentication, the application will run with the permissions of the currently logged in user or the Azure DevOps pipeline's permissions. Make sure that you are logged in to the proper Azure account using the [AZ command line interface tool](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest).
+
+## Configuration
+
+The configuration for the automation component is kept in [appsettings.json](/src/IotTelemetrySimulator.Automation/appsettings.json) or for local development purposes in **appsettings.Development.json**. It contains the following fields
+
+|Parameter|Description|Remarks
```

Comment:
The table in GH is cut. Wonder if adding breaking lines to some of the 2nd column items can improve it.

Reply:
Changed to 2 columns. I couldn't get it to work otherwise.